File: Week_01/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int count = 0;
        for (int start=0; count < nums.length; start++){
            int current = start;
            int prev = nums[start];
            do{
                current = (current + k)%nums.length;
                int tem = nums[current];
                nums[current] = prev;
                prev = tem;
                count ++;
            }while(start != current);
        }
    }
}

File: Week_01/rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        tem_nums = nums[n-k%n:]+ nums[:n-k%n]
        for i in range(n):
            nums[i] = tem_nums[i]